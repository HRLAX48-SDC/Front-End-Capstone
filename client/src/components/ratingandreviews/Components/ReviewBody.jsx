import React, { useState, useEffect} from 'react';

function ReviewBody (props) {



    return (
        <div>
            <div>{props.review.body}</div>
        </div>
    )

}
export default ReviewBody;