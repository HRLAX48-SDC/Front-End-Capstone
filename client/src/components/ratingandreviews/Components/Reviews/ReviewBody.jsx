import React, { useState, useEffect} from 'react';
import { Card, Button, Modal, Image, Row } from 'react-bootstrap'
import Images from './Images.jsx';

function ReviewBody (props) {
  const slicedBody = props.review.body.slice(0, 250);
  const reviewPhotos = props.review.photos;
  const [body, setBody] = useState(slicedBody);

  
  const showMore = () => {
    setBody(props.review.body)
  }


  return (
    <Card>
      <Card.Body>
        <Card.Text>
          {body}
        </Card.Text>
          {props.review.body.length > 250 &&
          <Button variant="outline-secondary"onClick={() => showMore()}>Show More</Button>
          }
          <Row>
            {reviewPhotos.length > 0 &&
            <div>
              {reviewPhotos.map((image, index) => 
                  <Images image={image} key={index}/>
              )}
            </div>
            }
          </Row>
      </Card.Body>
    </Card>
  )
}
export default ReviewBody;


//optional extras:

//default to first 250 chars
  //show more button should only appear when there are more than 250 chars

//display a "..." when review is trimmed
  {/* {props.review.body.length > 250 ? <>{body}...</>
          : <>{body}</>} */}