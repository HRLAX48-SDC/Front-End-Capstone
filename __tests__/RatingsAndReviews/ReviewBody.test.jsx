import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ReviewBody from '../../client/src/components/ratingandreviews/Components/Reviews/ReviewBody.jsx'

describe('<ReviewBody />', () => {
    const review = {
        body: "Wow this product is amazing, 100/10 recommend. I own 1000 of these muahahaha. ",
        date: "2021-11-10T00:00:00.000Z",
        helpfulness: 6,
        photos: [],
        rating: 5,
        recommend: true,
        response: null,
        review_id: 1094674,
        reviewer_name: "BananaBoy123",
        summary: "Is this really the best purchase ever?"
    } 
    const slicedReview =review.body.slice(0, 250);
    it('Should correctly render the reviewBody', () => {
        const reviewBody = renderer
        .create(<ReviewBody review={slicedReview}/>)
        .toJSON();
        expect(reviewBody.length).toBeLessThan(250)

    });

});