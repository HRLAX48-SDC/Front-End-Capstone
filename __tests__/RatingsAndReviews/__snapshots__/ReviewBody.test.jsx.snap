// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewBody /> Should correctly render the reviewBody 1`] = `undefined`;
